rattr/cli/parser.py

```diff
@@ -58,39 +118,123 @@ def parse_arguments() -> Namespace:
     )
 
     # TODO Allow user to add to this (will need config to respect this)
-    ARGUMENT_GROUP_PARSERS = (
+
+    # only these should be configurable in .toml file (filter and file )
+    # should come from cli args
+    TOML_ARG_GROUP_PARSERS = (
         FollowImports,
         ExcludeImports,
         ExcludePatterns,
         ShowWarnings,
         ShowPath,
         StrictOrPermissive,
         Output,
-        FilterString,
-        File,
         Cache,
     )
 
-    for argument_group_parser in ARGUMENT_GROUP_PARSERS:
-        parser = argument_group_parser.register(parser)
+    CLI_ARG_GROUP_PARSERS = TOML_ARG_GROUP_PARSERS + (FilterString, File)
+
+    for argument_group_parser in CLI_ARG_GROUP_PARSERS:
+        cli_parser = argument_group_parser.register(cli_parser)
+        if argument_group_parser in TOML_ARG_GROUP_PARSERS:
+            toml_parser = argument_group_parser.register(toml_parser)
 
     # TODO
     #   One moved to Python 3.9+ add `exit_on_error=False` to the
     #   `ArgumentParser` constructor and catch the error the same as below --
     #   this will give consistent behaviour between parser and validator errors
-    arguments = parser.parse_args()
 
-    for argument_group_parser in ARGUMENT_GROUP_PARSERS:
+    # extracting .toml expected field lists from 'TOML_ARG_GROUP_PARSERS'
+    toml_expected_fields: Tuple[str] = ()
+    for arg_group_parser in TOML_ARG_GROUP_PARSERS:
+        toml_expected_fields += tuple(
+            arg_group_parser.TOML_ARG_NAME_ARG_TYPE_MAP.keys()
+        )
+
+    if not project_toml_cfg:
         try:
-            arguments = argument_group_parser.validate(parser, arguments)
+            project_toml_cfg = load_cfg_from_project_toml()
+        except TOMLDecodeError as e:
+            # TODO: maybe construct a more informative message.
+            if exit_on_error:
+                error.fatal("Error parsing pyproject.toml file.")
```

Comment:
@BPHarris  maybe here we should just throw the original error with the original trace (+ maybe a pointer/path to the  .toml file)?